activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb

```diff
@@ -414,7 +414,7 @@ def default_index_type?(index) # :nodoc:
         def translate_exception(exception, message)
           return exception unless exception.respond_to?(:result)
 
-          case exception.result.try(:error_field, PGresult::PG_DIAG_SQLSTATE)
+          case exception.result.try(:error_field, PG::PG_DIAG_SQLSTATE)
```

Comment:
Of course, `PG_DIAG_SQLSTATE` is first defined as [PG::Constants::PG_DIAG_SQLSTATE](https://github.com/ged/ruby-pg/blob/master/ext/pg.c#L551) but `PG::Constants` is included into [PG](https://github.com/ged/ruby-pg/blob/master/ext/pg.c#L643) , [PG::Result](https://github.com/ged/ruby-pg/blob/master/ext/pg_result.c#L1216) and [PG::Connection](https://github.com/ged/ruby-pg/blob/master/ext/pg_connection.c#L3925) . Each of these would work.